File: Brainz/Model/Artist.cs
namespace brainz.Model
{
    using System;
    using Newtonsoft.Json;

    public partial class Artist
    {
        [JsonProperty("id")]
        public Guid Id { get; set; }

        [JsonProperty("type", NullValueHandling = NullValueHandling.Ignore)]
        public string Type { get; set; }

        [JsonProperty("type-id", NullValueHandling = NullValueHandling.Ignore)]
        public Guid? TypeId { get; set; }

        [JsonProperty("score")]
        public long Score { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("sort-name")]
        public string SortName { get; set; }

        [JsonProperty("country", NullValueHandling = NullValueHandling.Ignore)]
        public string Country { get; set; }

        [JsonProperty("area", NullValueHandling = NullValueHandling.Ignore)]
        public Area Area { get; set; }

        [JsonProperty("begin-area", NullValueHandling = NullValueHandling.Ignore)]
        public Area BeginArea { get; set; }

        [JsonProperty("disambiguation", NullValueHandling = NullValueHandling.Ignore)]
        public string Disambiguation { get; set; }

        [JsonProperty("life-span")]
        public ArtistLifeSpan LifeSpan { get; set; }

        [JsonProperty("tags", NullValueHandling = NullValueHandling.Ignore)]
        public Tag[] Tags { get; set; }

        [JsonProperty("aliases", NullValueHandling = NullValueHandling.Ignore)]
        public Alias[] Aliases { get; set; }

        [JsonProperty("gender", NullValueHandling = NullValueHandling.Ignore)]
        public string Gender { get; set; }
    }
}

File: go/src/model/artist.go
package model

import "fmt"

type Artist struct {
	ID             string   `json:"id"`
	Type           string   `json:"type,omitempty"`
	TypeID         string   `json:"type-id,omitempty"`
	Score          int      `json:"score"`
	Name           string   `json:"name"`
	SortName       string   `json:"sort-name"`
	Country        string   `json:"country,omitempty"`
	Area           Area     `json:"area,omitempty"`
	BeginArea      Area     `json:"begin-area,omitempty"`
	Disambiguation string   `json:"disambiguation,omitempty"`
	LifeSpan       Lifespan `json:"life-span"`
	Tags           []Tag    `json:"tags,omitempty"`
	Aliases        []Alias  `json:"aliases,omitempty"`
	Gender         string   `json:"gender,omitempty"`
}

func (artist Artist) DisambiguatedName() string {
	disambiguation := ""

	if artist.Disambiguation != "" {
		disambiguation = fmt.Sprintf(" (%s)", artist.Disambiguation)
	}

	return fmt.Sprintf("%s%s", artist.Name, disambiguation)
}